pagopa/io-app
# feat: [IOCOM-879] Bottom sheet for SEND's F24 with the new DS

This PR depends on #5156 

## Short description
This PR adds the bottom sheet to the SEND's message details, showing the F24 list.
![Simulator Screenshot - iPhone 15 - 2024-03-18 at 17 29 18](https://github.com/pagopa/io-app/assets/5150343/ef75099d-9c54-42db-914a-e6ab20351519)

## List of changes proposed in this pull request
- New `F24sWithBottomSheet`component to include and populate the F24 bottom sheet when there is more than one F24
- `F24` component links and renders the new `F24sWithBottomSheet` component
- `useAttachmentDownload` and `MessageDetailsAttachmentItem` accept a new callback to be called before screen navigation (in order to dismiss the bottom sheet after F24 selection)
- Previous `MessageAttachment` renamed to `LegacyMessageAttachment`

## How to test
Using the io-dev-api-server, generate a SEND message with multiple F24 attachments and check that the bottom sheet shows them properly.
